crates/autopilot/src/on_settlement_event_updater.rs

```diff
@@ -194,11 +195,16 @@ impl Inner {
         let effective_gas_price = receipt
             .effective_gas_price
             .with_context(|| format!("no effective gas price {hash:?}"))?;
-        let auction_external_prices = Postgres::get_auction_prices(ex, auction_id)
+        let auction_external_prices = self
+            .persistence
+            .auction_prices(auction_id)
             .await
             .with_context(|| {
                 format!("no external prices for auction id {auction_id:?} and tx {hash:?}")
-            })?;
+            })?
+            .into_iter()
+            .map(|(token, price)| (token.0, price.get().into()))
+            .collect::<BTreeMap<_, _>>();
```

Comment:
temporary code until I merge everything from https://github.com/cowprotocol/services/pull/2436

the whole `fetch_auction_data` function will be deleted.